### PR TITLE
feat(ci): Sonar Action to work with SonarCloud

### PR DESCRIPTION
**Pull request by sethkfman**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

The current Sonar action points to a self-host Sonar instance. This update move the Sonar Analysis to the Consensys SonarCloud. 

**Screenshots/Recordings**

NA

**Issue**

Progresses #???

**Checklist**

* [NA] There is a related GitHub issue
* [NA] Tests are included if applicable
* [NA] Any added code is fully documented

### CODE REVIEW

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -4,17 +4,19 @@ on:
...
-          fetch-depth: 0
-      - uses: sonarsource/sonarqube-scan-action@427bad70165a74f9d7133d48ea31b482a53ee9fa
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@5875562561d22a34be0c657405578705a169af6c
         with:
           args: >
             -Dsonar.javascript.lcov.reportPaths=tests/coverage/lcov.info
         env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
```
[the remark this answers is not in the document]
Comment (sethkfman) [reply .github/workflows/sonar.yml:20]: Not at this time, can we make that change across all actions in a later PR?

[the remark this answers is not in the document]
Comment (sethkfman) [reply .github/workflows/sonar.yml]: So from reading the docs this doesn't need to explicitly be added. I have removed `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}`

**Review by legobeat — DISMISSED** [review]
Left two nits and did not verify that this actually runs but looks good to go otherwise

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**